### DISCUSSION TITLE
Can't login in any deployment env, only localhost

### DISCUSSION THREAD

**Discussion by XaviGIT** [vercel/community#4609] [Help]
### Summary

I'm using Next.js with Supabase template and didn't changed the logic of the authentication.
Env variables are set on the project and active to all envs.
Can't find details about the error, only that it failed. 

In localhost works just fine.

### Steps to Reproduce

Code for the sign in:

`import { createRouteHandlerClient } from '@supabase/auth-helpers-nextjs'
import { cookies } from 'next/headers'
import { NextResponse } from 'next/server'

export const dynamic = 'force-dynamic'

export async function POST(request: Request) {
  const requestUrl = new URL(request.url)
  const formData = await request.formData()
  const email = String(formData.get('email'))
  const password = String(formData.get('password'))
  const supabase = createRouteHandlerClient({ cookies })

  const { error } = await supabase.auth.signInWithPassword({
    email,
    password,
  })

  if (error) {
    return NextResponse.redirect(
      `${requestUrl.origin}/login?error=Could not authenticate user`,
      {
        // a 301 status is required to redirect from a POST to a GET route
        status: 301,
      }
    )
  }

  return NextResponse.redirect(`${requestUrl.origin}/dashboard`, {
    // a 301 status is required to redirect from a POST to a GET route
    status: 301,
  })
}
`

**Comment by amyegan** [comment]
Do you have your Supabase settings configured for connection with your production domain? I believe Supabase sets it to `localhost:3000` by default, so you would need to go into your Supabase project's Authentication section, open the URL Configuration tab, and update the Site URL.

**Comment by XaviGIT** [comment]
Hey Amy,

I have the root domain configured in Site URL and I also added https://*.
vercel.app to my redirect URL's.
Still having the same issue. Any idea on what am I doing wrong?

Thanks

Amy ***@***.***> escreveu no dia quarta, 25/10/2023 à(s)
21:33:
